### PR TITLE
fix(i): Handle field in view query but not view SDL

### LINKED ISSUE #2183: Schema View: panics with field in SDL that partially matches aggregate in query
Specified in SDL "count" field and in query "_count". Upon querying it panics:
```go
func TestView_UnknownFieldInView(t *testing.T) {
	test := testUtils.TestCase{
		Description: "One to many view with count",
		Actions: []any{
			testUtils.SchemaUpdate{
				Schema: `
					type Author {
						name: String
						books: [Book]
					}
					type Book {
						name: String
						author: Author
					}
				`,
			},
			testUtils.CreateView{
				Query: `
					Author {
						name
						_count(books: {})
					}
				`,
				SDL: `
					type AuthorView {
						name: String
						count: Int
					}
				`,
			},
			// bae-ef9cd756-08e1-5f23-abeb-7b3e6351a68d
			testUtils.CreateDoc{
				CollectionID: 0,
				Doc: `{
					"name":	"Harper Lee"
				}`,
			},
			testUtils.Request{
				Request: `query {
							AuthorView {
								name
							}
						}`,
			},
		},
	}

	testUtils.ExecuteTestCase(t, test)
}
```
Stacktrace:
```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 82 [running]:
testing.tRunner.func1.2({0x1032c8cc0, 0x14000042360})
	/Users/islam/sdk/go1.20.7/src/testing/testing.go:1526 +0x1c8
testing.tRunner.func1()
	/Users/islam/sdk/go1.20.7/src/testing/testing.go:1529 +0x384
panic({0x1032c8cc0, 0x14000042360})
	/Users/islam/sdk/go1.20.7/src/runtime/panic.go:884 +0x204
github.com/sourcenetwork/defradb/core.(*DocumentMapping).SetFirstOfName(...)
	/Users/islam/dev/sourcenetwork/defradb/core/doc.go:180
github.com/sourcenetwork/defradb/planner.(*viewNode).Value(0x14000447f10)
	/Users/islam/dev/sourcenetwork/defradb/planner/view.go:72 +0x23c
github.com/sourcenetwork/defradb/planner.(*selectNode).Next(0x1400061a5b0)
	/Users/islam/dev/sourcenetwork/defradb/planner/select.go:157 +0x60
github.com/sourcenetwork/defradb/planner.(*selectTopNode).Next(0x1400011ece8?)
	/Users/islam/dev/sourcenetwork/defradb/planner/select.go:70 +0x28
github.com/sourcenetwork/defradb/planner.(*Planner).executeRequest(0x18?, {0x103055da0?, 0x140004594f0?}, {0x1033d5c60, 0x14000791090})
	/Users/islam/dev/sourcenetwork/defradb/planner/planner.go:520 +0x48
github.com/sourcenetwork/defradb/planner.(*Planner).RunRequest(0x140001b6af8?, {0x1033c9130, 0x1400019e018}, 0x140004594f0)
	/Users/islam/dev/sourcenetwork/defradb/planner/planner.go:575 +0x230
github.com/sourcenetwork/defradb/db.(*db).execRequest(0x14000109080, {0x1033c9130?, 0x1400019e018}, {0x102b13e84, 0x39}, {0x1033d9568?, 0x140009b9730})
	/Users/islam/dev/sourcenetwork/defradb/db/request.go:53 +0x3c8
github.com/sourcenetwork/defradb/db.(*implicitTxnDB).ExecRequest(0x140000ca028, {0x1033c9130, 0x1400019e018}, {0x102b13e84, 0x39})
	/Users/islam/dev/sourcenetwork/defradb/db/txn_db.go:45 +0x150
github.com/sourcenetwork/defradb/tests/integration.executeRequest(0x14000439340, {{0x0, 0x0}, {0x0, 0x0}, {0x102b13e84, 0x39}, {0x0, 0x0, 0x0}, ...})
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:1527 +0x11c
github.com/sourcenetwork/defradb/tests/integration.performAction(0x14000439340?, 0x0?, {0x1032d9cc0?, 0x140006fdf80?})
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:307 +0x5bc
github.com/sourcenetwork/defradb/tests/integration.executeTestCase({0x1033c9130?, 0x1400019e018}, 0x140000a2ea0, {0x140000c0340, 0x2, 0x2}, {{0x102adaf4b, 0x1b}, {0x1400097cf00, 0x6, ...}, ...}, ...)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:206 +0x804
github.com/sourcenetwork/defradb/tests/integration.ExecuteTestCase(0x1032d9cc0?, {{0x102adaf4b, 0x1b}, {0x1400097cf00, 0x6, 0x6}, {0x0, {0x0, 0x0, 0x0}}})
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:162 +0x580
github.com/sourcenetwork/defradb/tests/integration/view/one_to_many.TestView_UnknownFieldInView(0x0?)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/view/one_to_many/with_alias_test.go:382 +0x350
testing.tRunner(0x140000a2ea0, 0x1033ad2d8)
	/Users/islam/sdk/go1.20.7/src/testing/testing.go:1576 +0x10c
created by testing.(*T).Run
	/Users/islam/sdk/go1.20.7/src/testing/testing.go:1629 +0x368
FAIL	github.com/sourcenetwork/defradb/tests/integration/view/one_to_many	0.877s
FAIL
```

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #2183 #2189

## Description

Handle field in view query but not view SDL.

As such fields cannot be queried by the user anyway, we can safely ignore them (instead of panicing).

### CODE REVIEW

File: tests/integration/view/one_to_many/simple_test.go [hunk 1]
```diff
@@ -239,3 +239,67 @@ func TestView_OneToManyOuterToInnerToOuter_Errors(t *testing.T) {
 
 	testUtils.ExecuteTestCase(t, test)
 }
+
+func TestView_OneToManyWithRelationInQueryButNotInSDL(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "One to many view",
```
Comment (islamaliev) [inline tests/integration/view/one_to_many/simple_test.go:245]: todo: please the description more descriptive :)

File: tests/integration/view/simple/simple_test.go [hunk 2]
```diff
@@ -160,3 +160,101 @@ func TestView_SimpleWithFieldSubset_ErrorsSelectingExcludedField(t *testing.T) {
...
+				Doc: `{
+					"name":	"John"
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+							UserView {
+								name
+							}
+						}`,
```
Comment (islamaliev) [inline tests/integration/view/simple/simple_test.go:249]: nitpick: indentation here (and in above query) is a bit off

**Review by islamaliev — APPROVED** [review]
Look good. Have just minor comments

File: tests/integration/view/one_to_many/simple_test.go (hunk 1 above)
Comment (AndrewSisley) [reply tests/integration/view/one_to_many/simple_test.go:245]: Lol - sorry :grin: 

- [ ] Sort out description(s?)

File: tests/integration/view/simple/simple_test.go (hunk 2 above)
Comment (AndrewSisley) [reply tests/integration/view/simple/simple_test.go:249]: cheers, will fix

- [ ] Fix indentation

**Pull request merged**